src/pairs/Player.py

```diff
@@ -16,4 +19,8 @@ def reset(self):
         for card_type in self.hand_state:
             self.hand_state[card_type] = 0
 
+    def copy(self):
```

Comment:
Rename to `deepcopy` and use `self.hand_state.deepcopy()`, etc, for the variables you're passing in.

Reply:
done